assets/app/scripts/controllers/overview.js

```diff
@@ -69,6 +73,23 @@ angular.module('openshiftConsole')
       Logger.log("services (list)", $scope.services);
     }));
 
+    watches.push(DataService.watch("routes", $scope, function(routes) {
+      var routeMap = $scope.routesByService;
+      var displayRouteMap = $scope.displayRouteByService;
```

Comment:
reinitialize the scope vars in the callback, otherwise you won't catch deletions